knowledge-base/spreadsheet-get-cell-values.md

```diff
@@ -0,0 +1,210 @@
+---
+title: Iterate Spreadsheet Data and Get Cell Values
+description: Learn how to iterate the Spreadsheet data cells and get their values as objects or strings.
+type: how-to
+page_title: How to Iterate Spreadsheet Data and Get Cell Values
+slug: spreadsheet-kb-get-cell-values
+tags: telerik, blazor, spreadsheet
+ticketid: 1661533
+res_type: kb
+---
+
+## Environment
+
+<table>
+    <tbody>
+        <tr>
+            <td>Product</td>
+            <td>Spreadsheet for Blazor</td>
+        </tr>
+    </tbody>
+</table>
+
+## Description
+
+This KB article answers the following questions:
+
+* How to iterate Spreadsheet cells?
+* How to get the Spreadsheet cell values?
+* How to obtain the data that users entered in the Blazor Spreadsheet?
+
+## Solution
+
+1. Install the required [Telerik Document Processing NuGet packages](https://docs.telerik.com/devtools/document-processing/libraries/radspreadprocessing/getting-started):
+    * `Telerik.Documents.Spreadsheet`
+    * `Telerik.Documents.Spreadsheet.FormatProviders.OpenXml`
+1. Set the [`@ref` attribute of the Spreadsheet]({%slug spreadsheet-overview%}#spreadsheet-reference-and-methods), so that you can call component methods.
+1. Use the [`ExportToExcelAsync()` Spreadsheet method]({%slug spreadsheet-overview%}#spreadsheet-reference-and-methods) to get the Excel file as a `byte[]`.
+1. Create a `MemoryStream` that contains the Excel file byte array.
+
+    The steps below use the [Telerik RadSpreadProcessing API](https://docs.telerik.com/devtools/document-processing/api/) and are outside the scope of Telerik UI for Blazor.
```

Comment:
I am not sure that this is rendering as expected (or the best possible way). I played around a bit with it, but the best I got was an info note and a restarted list of steps (less than ideal). You can experiment some more but we might be stuck with the current solution :(

![image](https://github.com/user-attachments/assets/2f02adb7-bbfc-49b7-bc6e-6483eef744f9)

Reply:
Yes, I also couldn't come up with anything better.